File: absolute-zero/examples/csharp/Nop.cs
/**
 * Certified Null Operation in C#
 *
 * A program that does absolutely nothing at the application level.
 * Exits with code 0 (success) without any observable side effects.
 *
 * Properties:
 * - Terminates immediately
 * - No I/O operations
 * - No memory allocation (beyond CLR overhead)
 * - Exit code 0
 *
 * Compile: csc Nop.cs
 * Run: ./Nop.exe (or Nop on Unix with Mono/dotnet)
 */

using System;

class Nop
{
    static void Main(string[] args)
    {
        // Empty Main - the minimal CNO in C#
        // The CLR handles startup/shutdown, but at the
        // application logic level, this computes nothing observable.
    }
}

/*
 * Verification notes:
 * - CLR loads assemblies and initializes the runtime
 * - Garbage collector is initialized but does no work
 * - JIT compilation may occur
 * - At application level: CNO
 * - At runtime level: significant infrastructure
 *
 * This demonstrates the abstraction gap between managed code
 * and the underlying runtime environment.
 */

File: absolute-zero/examples/go/nop.go
/*
 * Certified Null Operation in Go
 *
 * A program that does absolutely nothing at the application level.
 * Exits with code 0 (success) without any observable side effects.
 *
 * Properties:
 * - Terminates immediately
 * - No I/O operations
 * - No goroutines spawned
 * - No channel operations
 * - Exit code 0
 *
 * Build: go build nop.go
 * Run: ./nop
 */

package main

func main() {
	// Empty main - the minimal CNO in Go
	// The Go runtime handles initialization and cleanup,
	// but at the application level, this computes nothing observable.
}

/*
 * Verification notes:
 * - Go runtime initializes (scheduler, garbage collector, etc.)
 * - Goroutine for main() is created
 * - Memory allocator is initialized
 * - At application level: CNO
 * - At runtime level: goroutine scheduling occurs
 *
 * This demonstrates the gap between application logic and
 * the concurrent runtime that Go provides.
 *
 * Interestingly, even this empty program benefits from Go's
 * concurrent runtime infrastructure, though no concurrency is used.
 */
